src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameProjection.java

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright @ 2019 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc.vp8;
+
+import org.jetbrains.annotations.*;
+import org.jitsi.impl.neomedia.codec.video.vp8.*;
+import org.jitsi.impl.neomedia.rtp.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+import org.jitsi.videobridge.cc.*;
+
+import java.util.*;
+
+/**
+ * Represents a VP8 frame projection. It puts together all the necessary bits
+ * and pieces that are useful when projecting an accepted VP8 frame. A
+ * projection is responsible for rewriting a VP8 packet. Instances of this class
+ * are thread-safe.
+ *
+ * @author George Politis
+ */
+public class VP8FrameProjection
+{
+    /**
+     * The {@link Logger} to be used by this instance to print debug
+     * information.
+     */
+    private static final Logger
+        logger = Logger.getLogger(VP8FrameProjection.class);
+
+    /**
+     * The time (in millis) to wait before considering that this frame is fully
+     * projected (see {@link #isFullyProjected(long)}).
+     */
+    private final long WAIT_MS = 5000;
+
+    /**
+     * A timestamp of when this instance was created. It's used to compute how
+     * long this instance has lived. That's used to determines when it's time to
+     * forget about this frame projection (see {@link #isFullyProjected(long)}).
+     */
+    private final long createdMs;
+
+    /**
+     * The projected {@link VP8Frame}.
+     */
+    private final VP8Frame vp8Frame;
```

Comment:
this is the 'currently' projected frame i guess, yeah? or currently the frame in progress of being projected?

Reply:
Yeah

Reply:
Actually, it's the frame that this projection is projecting. I'm not sure what you mean by "currently" projected frame, I don't think there's such notion. Maybe you mean the "last" projected frame? In any case, this doesn't really matter at this point, this is just the frame that this projection is supposed to project.